rfc/plugins-ui.md

```diff
@@ -1,115 +1,124 @@
+| **Author(s)** | Brandon Powell (`@bpowell`), Cameron Motevasselani (`@link108`), Clay Mccoy (`@claymccoy`) | Chris Thielen (`@christopherthielen`)
 | **SIG / WG** | Platform SIG |
 | **Obsoletes** | [Google Doc](https://docs.google.com/document/d/16WmRSziTJsSBZ1kuKUfVleLAYMIxmfvz/edit) |
 
 ## Motivation
-Just like the backend in Spinnaker, the frontend needs to be extensible for others to add more functionality. Deck should be able to be extended by well defined extension points. Some examples of extension points could be:
-1. Custom stages
-2. Modifying header and footer
-3. Custom search filters
-4. Custom routes
-5. Adding labels or other information to sections in the infrastructure tab
+Spinnaker is used by many companies, each having wildly different needs.  Many enterprise companies have heavily customized Spinnaker improve the experience based on their needs.  Some examples of extensions seen in the real world are: 
 
+1. Custom stages
+2. Modifying header and footer - show screenshot
+3. Custom search filters - show screenshot
+4. Custom routes - show screenshot
+5. Custom details panels - show screenshot
+6. Adding labels or other information to sections in the infrastructure tab
+ 
+However, the process of customization necessitates a custom build of the spinnaker services and UI.  Adding a build step greatly increases the operational costs and complexity of running spinnaker.  Smaller companies with fewer resources likely lack the engineering and ops bandwidth to do this.
+
+This proposal explores a world where the the UI (Deck) is extensible _without a build step_.  
+  
 This design doc references the [Plugins RFC](./plugins.md).
 
-This doc focuses on
-* How plugins are implemented in Deck
-* Deck  Plugin Interaction
+This doc focuses on:
+* [Loading Plugin code into Deck](#Loading-Plugin-code-into-Deck)
+* [Bootstrapping Deck](#Bootstraping-Deck)
+* [Extension points](#Extension-Points)
+* [Plugin development](#Plugin-development)
 
-## How Plugins are Implemented in Deck
-### Load Plugins into Deck
-In order for plugins to have a UI component, they must be loaded into Deck’s runtime. In order to load plugins to Deck’s runtime, Deck needs to know two pieces of information:
-* What plugins are enabled 
-* The location of the resources a plugin needs.
+## Loading Plugin code into Deck
 
-### Plugin Configuration via Front50
-In order for Deck to know which plugins it can load, it must have access to some configuration information, including the name of the plugin and where to download the plugin resource(s).
+### Gathering plugin metadata
+In order for Deck to know which plugins it can load, it must have access to some configuration information, including the name of the plugins that are enabled and where to download the plugin resources (JavaScript bundles, CSS, images, etc).  This Plugin Configuration will be fetched from Front50.
```

Comment:
```suggestion
In order for Deck to know which plugins it can load, it must have access to some configuration information, including the name of the plugins that are enabled and where to download the plugin resources (JavaScript bundles, CSS, images, etc).  This Plugin Configuration will be fetched from Front50 via Gate.
```